src/components/data/DataTable.tsx

```diff
@@ -83,6 +83,11 @@ type ComponentProps<Row> = Pick<
    */
   orderableColumns?: Array<keyof Row>;
 
+  /**
+   * The initial order direction to set when a column becomes the ordered one.
+   */
+  initialColumnsOrderDir?: Partial<Record<keyof Row, OrderDirection>>;
```

Comment:
The name is a little awkward. Maybe something like `initialOrderForColumn`. Another possible API direction would be to combine this with `orderableColumns` so you'd specify which columns can be used for sorting as well as the associated direction in one prop.

Reply:
> Another possible API direction would be to combine this with `orderableColumns` so you'd specify which columns can be used for sorting as well as the associated direction in one prop.

Aha! I actually really like this approach. I even tried to restrict keys in `initialOrderForColumn` to the values in `orderableColumns`, but failed. This would solve that problem.

I'll try to allow both an array or a record to be passed to `orderableColumns`, for backwards compatibility and because it's more convenient if you don't want change the default direction, but let's see if that doesn't cause any problems.